### DISCUSSION TITLE
Help configuring a wakeup light alarm clock (TuyaMCU)

### DISCUSSION THREAD

**Discussion by gado4525** [arendst/Tasmota#21107] [Support request]
![61fGT01CsuL _AC_SL1498_](https://github.com/arendst/Tasmota/assets/92314530/9db24410-4d14-41b3-8673-9c791b6af6d0)

First time experience, I've managed to flash tasmota 13.4.0 into a generic "wake up light" alarm clock that has a tuya TYWE3S chip inside.  The device doesn't have a template and similar devices templates won't work. The only way to control it is using TuyaMCU.
 These are the device's dpId and functions:
 
![Screenshot 2024-04-05 133252](https://github.com/arendst/Tasmota/assets/92314530/ed3a07a4-1f25-4e80-9d0c-845a494b1e69)

For example dpId 101 controls the light on off, and if I map TuyaMCU 11,101 I got the first working relay. Once turned on the device has 1-20 steps of brightness controlled by dpID 102. I once again map TuyaMCU 21,102 for the first dimmer. Set DimmerRange 0-20, this way when testing "dimmer 50", the light is correctly set at step 10 (it shows in the device screen).  The first problem arise here. The relay 1 and the dimmer doesn't like each other. If I move the slider being the relay in OFF state, it changes the device state to ON but not turning on the light. I have to put the slider in 0 to change again the state to OFF then I can correctly use relay1 to turn on the light and this will be turned into the last step set (like 20-maximum) and the slider will remaing in 0. Sometimes relay 1 doesn't want to turn off the light unless brightness is set to maximum. 
The second function Im trying to configure is with dpId 104. This can control the device screen brightness in 3 steps (0-1-2) being 0 completely off. I can correctly test with TuyaSend2 104,0 to turn off clock brightness. dpId 104 uses integer type. How do I configure another slider or a button that can cycle between 0-1-2?  
Can someone with more experience help me figure this out and after maybe help me create a template for this and similar devices?

**Comment by marto68** [comment]
Hi, I purchased one myself and have just taken it apart. It had a cb3s chip instead.
I want to flash tuya like you did.
You said you used tuya convert, was that   over wifi or using a serial converter?
Also did you get it all working?
Any ideas would be appreciated.
Thanks
Marty

**Reply by sfromis** [reply]
The BK7231N chip inside the CB3S module is not compatible with Tasmota. For that, you'd have to replace the CB3S module, many exists with the same form factor, with a small variation of Espressif chips.

ESP-12 ESP-12H ESP-C3-12F ESP8684-WROOM-01C ESP8684-WROOM-04C ESP8685-WROOM-01 ESP8685-WROOM-04 ESPC2-12 wireless_tag_WT018684-S5 wireless_tag_WT32C3-S5

**Reply by gado4525** [reply]
> Hi, I purchased one myself and have just taken it apart. It had a cb3s chip instead. I want to flash tuya like you did. You said you used tuya convert, was that over wifi or using a serial converter? Also did you get it all working? Any ideas would be appreciated. Thanks Marty

Hello Marty, I did flash it using a serial device and soldering but never got it completely working and no one ever tried to help here or tasmota discord to get a valid configuration so I ended up giving up and just stored the little light.
